api/api/application/services/dataset_validation.py

```diff
@@ -106,6 +106,9 @@ def dataset_has_correct_data_types(
         data_frame,
     )
     for column in schema.columns:
+        if column.name not in column_types:
```

Comment:
I don't understand why we are matching the column name with the column type? Wouldn't we expect the column datatype to match the column_type?

If we then make that change this function would be a bit redundant as every column would always pass.

Reply:
I check here if column.name key is in column_types dictionary. I modified extract_athena_types function to not include that key if column is empty.

Reply:
I used API to test it. I first create schema and then upload it.
If you try to create schema from dataframe with empty column, it by default infers double, I guess.